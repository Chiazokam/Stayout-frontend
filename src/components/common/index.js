import BackImage from './BackImage';
import Logo from './Logo';
import Container from './Container';
import MobileHeader from './MobileHeader';
import Hamburger from './Hamburger';
import Button from './Button';
import CustomSection from './CustomSection';
import ThinLine from './ThinLine';
import Dropdown from './Dropdown';

export {
  BackImage,
  Logo,
  Container,
  MobileHeader,
  Hamburger,
  Button,
  CustomSection,
  ThinLine,
  Dropdown
};
