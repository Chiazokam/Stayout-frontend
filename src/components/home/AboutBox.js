import React from 'react';
import { Row, Col } from 'reactstrap';


const AboutBox = (props) => {
  const {
    className, src, alt, aboutText
  } = props;

  return (
    <Col md='4' className={`about-box ${className}`}>
      <div>
        <img className='about-image' src={src} alt={alt} />
        <p className='about-text'>{aboutText}</p>
      </div>
    </Col>
  );
};

export default AboutBox;
