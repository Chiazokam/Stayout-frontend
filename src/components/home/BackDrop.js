import React from 'react';

const BackDrop = () => (
  <section className='home-backdrop' />
);

export default BackDrop;
