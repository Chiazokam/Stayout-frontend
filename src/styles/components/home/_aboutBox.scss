.home-about {
  position: absolute;
  display: flex;
  top: 38rem;
  left: 3.5%;
  width: 87%;
  padding-left: 7%;
}

.about-box{
  flex: 2;
  color: $off-white;
  text-align: center;
}

.about-box-left {
  padding-right: 2rem;
}

.about-box-middle {
  padding: 0 4rem;
}

.about-box-right {
  padding-left: 4rem;
}

.about-image {
  height: 73px;
  margin-right: 1.5rem;
}

.about-text {
  font-size: 24px;
  letter-spacing: 1px;
  font-weight: 100;
}

.col-md-4 {
  max-width: 40.33333%;
}

@media only screen and (max-width: 767px) {
  .home-about {
    top: 10%;
    left: 0;
    flex-direction: column;
    width: 92%;
  }

  .about-box {
    margin-top: 9%;
  }

  .row {
    display: grid;
    margin: 0 auto;
  }

  .col-md-4 {
    max-width: 100.33333%;
  }

  .about-box-right,
  .about-box-middle,
  .about-box-left {
    padding: 0 4rem;
  }
}

@media only screen and (max-width: 414px) {
  .about-text {
    font-size: 20px;
  }

  .home-about {
    top: 12%;
  }
}

@media only screen and (max-width: 375px) {
  .about-box {
    margin-top: 6%;
  }
}
