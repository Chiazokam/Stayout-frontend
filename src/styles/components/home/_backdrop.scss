.home-backdrop {
  background-color: $dark-brown;
  height: 86vh;
  position: relative;
}

@media only screen and (max-width: 767px) {
  .home-backdrop {
    display: none;
  }
}
