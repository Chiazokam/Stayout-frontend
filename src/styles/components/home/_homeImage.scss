.home-image {
  background-image: linear-gradient(rgba(0,0,0,0), rgba(0,0,0,0.9)), url('https://res.cloudinary.com/zokky/image/upload/v1563640393/stayout.jpg');
  height: 88%;
  background-repeat: no-repeat;
  background-size: cover;
  position: absolute;
  top: 5rem;
  left: 4.5%;
  width: 91.5%;
}

@media only screen and (max-width: 1160px) {
  .home-image {
    height: 102%;
  }
}

@media only screen and (max-width: 767px) {
  .home-image {
  background-image: linear-gradient(rgba(0,0,0,0.4), rgba(0,0,0,0.9)), url('https://res.cloudinary.com/zokky/image/upload/v1563640393/stayout.jpg');
    height: 100%;
    top: 4rem;
    left: 0;
    width: 100%;
  }
}