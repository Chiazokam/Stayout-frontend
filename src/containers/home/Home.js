import React, { Component } from 'react';
import { Row } from 'reactstrap';
import Media from 'react-media';
import { BackDrop, AboutBox } from '../../components/home';
import user from '../../assets/images/user.svg';
import team from '../../assets/images/team.svg';
import card from '../../assets/images/card.svg';
import {
  BackImage,
  Logo,
  MobileHeader,
  Hamburger,
  Button,
  CustomSection,
  ThinLine,
  Dropdown
} from '../../components/common';

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isMenuOpen: false,
      menu: ['Sign in']
    };
  }

  toggleMenu = () => {
    this.setState((prevState, props) => ({
      isMenuOpen: !prevState.isMenuOpen,
    }));
  }

  render() {
    const { isMenuOpen, menu } = this.state;
    let dropdownMenu = 'dropdown-close';
    if (isMenuOpen) {
      dropdownMenu = 'dropdown-open';
    }

    return (
      <React.Fragment>
        <BackDrop />
        <Logo className='logo' />
        <h3 className='nav-web-home'> Sign in </h3>

        <MobileHeader className='mobile-header'>
          <Logo className='logo-mobile' />
          <Hamburger
            clicked={this.toggleMenu}
          />
        </MobileHeader>
        <BackImage className='home-image' />

        <Button
          className='home-button'
        >
          Search Spaces
        </Button>
        <CustomSection className='home-about'>
          <Row>
            <AboutBox
              className='about-box-left'
              src={user}
              alt='Single User'
              aboutText='Find your workspace conveniently'
            />
            <ThinLine className='home-thin-line' />
            <AboutBox
              className='about-box-middle'
              src={team}
              alt='Team'
              aboutText='Find a workspace for your team'
            />
            <ThinLine className='home-thin-line' />
            <AboutBox
              className='about-box-right'
              src={card}
              alt='Pay Online'
              aboutText='Book and make your payments online'
            />
          </Row>
        </CustomSection>

        <Media query='(max-width: 767px)'>
          {matches => (matches ? (
            <Dropdown
              menu={menu}
              className={`home-dropdown ${dropdownMenu}`}
            />
          ) : (
            null
          ))
          }
        </Media>

      </React.Fragment>
    );
  }
}

export default Home;
