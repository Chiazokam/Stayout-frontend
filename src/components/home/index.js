import BackDrop from './BackDrop';
import AboutBox from './AboutBox';

export { BackDrop, AboutBox };
