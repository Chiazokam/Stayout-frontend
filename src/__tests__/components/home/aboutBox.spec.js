import React from 'react';
import { shallow } from 'enzyme';
import { AboutBox } from '../../../components/home';

describe('AboutBox Tests', () => {
  it('should render correctly', () => {
    const tree = shallow(<AboutBox />);
    expect(tree).toMatchSnapshot();
  });
});
