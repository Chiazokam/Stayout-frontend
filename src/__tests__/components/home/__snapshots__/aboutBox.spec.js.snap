// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutBox Tests should render correctly 1`] = `ShallowWrapper {}`;
